iree/compiler/Dialect/Flow/Transforms/RematerializeDispatchConstants.cpp

```diff
@@ -144,18 +173,20 @@ LogicalResult inlineDispatchRegionOperandsUsingValue(
   // Remove the dispatch region args and the block args that have been
   // replaced.
   for (unsigned argIndex : llvm::reverse(argIndices)) {
-    dispatchRegionOp.getOperation()->eraseOperand(
-        dispatchRegionOp.mapArgOperandToOpOperand(argIndex));
+    dispatchOp.getOperation()->eraseOperand(
+        dispatchOp.mapArgOperandToOpOperand(argIndex));
     entryBlock.eraseArgument(argIndex);
   }
+  modifyOperandSegmentSizeAttr(context, dispatchOp, argIndices.size());
 
   return success();
 }
 
 // Rematerializes a constant inside of all dispatch regions that use it.
-// Afterward the constant is only removed if there are no other uses within the
-// non-dispatch block (such as by sequencer ops).
-LogicalResult rematerializeConstantInDispatchRegions(ConstantOp constantOp) {
+// Afterward the constant is only removed if there are no other uses within
+// the non-dispatch block (such as by sequencer ops).
+LogicalResult rematerializeConstantInDispatchRegions(ConstantOp constantOp,
+                                                     MLIRContext *context) {
```

Comment:
nit: you can get a context from any type - like `constantOp.getContext()` - MLIR tends to omit the context arg unless required (for example, IntegerAttr::get requires it because APInt has no context, but DenseElementsAttr::get doesn't because it takes a ShapedType where it can getContext())

Reply:
Done, I thought only ops with blocks (e.g FuncOp) have it. thanks!